tests/lib/nested.sh

```diff
@@ -598,6 +598,67 @@ nested_ensure_ubuntu_save() {
     fi
 }
 
+prepare_snapd() {
+    if [ "$NESTED_BUILD_SNAPD_FROM_CURRENT" = "true" ]; then
+        echo "Repacking snapd snap"
+        local snap_name output_name snap_id
+        if nested_is_core_16_system; then
+            snap_name="core"
+            output_name="core-from-snapd-deb.snap"
+            snap_id="99T7MUlRhtI3U0QFgl5mXXESAiSwt776"
+        else
+            snap_name="snapd"
+            output_name="snapd-from-deb.snap"
+            snap_id="PMrrV4ml8uWuEUDBT8dSGnKUYbevVhc4"
+        fi
+
+        "$TESTSTOOLS"/snaps-state repack_snapd_deb_into_snap "$snap" "$NESTED_ASSETS_DIR"
+        mv "$NESTED_ASSETS_DIR/$output_name" "$(nested_get_extra_snaps_path)/$output_name"
+
+        # sign the snapd snap with fakestore if requested
+        if [ "$NESTED_SIGN_SNAPS_FAKESTORE" = "true" ]; then
+            make_snap_installable_with_id --noack "$NESTED_FAKESTORE_BLOB_DIR" "$(nested_get_extra_snaps_path)/$output_name" "$snap_id"
+        fi
+    fi
+}
+
+prepare_kernel() {
+    # allow repacking the kernel
+    if [ "$NESTED_REPACK_KERNEL_SNAP" = "true" ]; then
+        echo "Repacking kernel snap"
+        local kernel_snap output_name snap_id
+        output_name="pc-kernel.snap"
+        snap_id="pYVQrBcKmBa0mZ4CCN7ExT6jH8rY1hza"
+
+        if nested_is_core_16_system || nested_is_core_18_system; then
+            kernel_snap=pc-kernel-new.snap
+            repack_kernel_snap "$kernel_snap"
+        elif nested_is_core_20_system || nested_is_core_22_system; then
+            snap download --basename=pc-kernel --channel="$VERSION/edge" pc-kernel
+
+            # set the unix bump time if the NESTED_* var is set,
+            # otherwise leave it empty
+            local epochBumpTime
+            epochBumpTime=${NESTED_CORE20_INITRAMFS_EPOCH_TIMESTAMP:-}
+            if [ -n "$epochBumpTime" ]; then
+                epochBumpTime="--epoch-bump-time=$epochBumpTime"
+            fi
+
+            uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$NESTED_ASSETS_DIR" "$epochBumpTime"
```

Comment:
nitpick: you don't need the comment, if you rewrite the code to be more explicit:
```suggestion
            local INITRAMFS_EXTRA_OPTS
            if [ -n "$NESTED_CORE20_INITRAMFS_EPOCH_TIMESTAMP" ]; then
                INITRAMFS_EXTRA_OPTS="--epoch-bump-time=$NESTED_CORE20_INITRAMFS_EPOCH_TIMESTAMP"
            fi
            # shellcheck disable=SC2086
            uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$NESTED_ASSETS_DIR" $INITRAMFS_EXTRA_OPTS
```
Note that in any case you should not use quotes around `"$epochBumpTime"` when you call `uc20_build_initramfs_kernel_snap`, or it will be passed an empty argument, which might cause problems.

Reply:
Ah, I see that you just moved this preexisting code around, so it's not anything new. Well, still consider if my comment makes sense.